### PR TITLE
feat: enable mock servers with intercept command

### LINKED ISSUE #1910: Enable quick start to run with local-sequencer
To help with different tutorials and guides, it would be useful to be able to run a custom sequencer with the quick start command. 

Currently if you try you get an address conflict:
```
rollkit start --rollkit.aggregator --rollkit.da_address http://localhost:7980
Error intercepting command:  Reading rollkit.toml: no rollkit.toml found
I[2024-11-05|14:40:59.826] Found private validator                      module=main keyFile=/Users/matt/.rollkit/config/priv_validator_key.json stateFile=/Users/matt/.rollkit/data/priv_validator_state.json
I[2024-11-05|14:40:59.826] Found node key                               module=main path=/Users/matt/.rollkit/config/node_key.json
I[2024-11-05|14:40:59.826] Found genesis file                           module=main path=/Users/matt/.rollkit/config/genesis.json
ERROR: failed to launch mock sequencing server: listen tcp 127.0.0.1:50051: bind: address already in use
```

I think we just need to add logic to check if there is already a sequencer running on the default port like we do with the DA server.

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

This PR does two key things:
1. Handles already running sequencer
2. Enables using mock services with the intercept command. 

1 is pretty straight forward as we did this for the DA server as well. The one thing to note though with this PR is that because we are checking for already running servers, I just remove the redundant check of if the flag was set. I add a comment to the code to explain the reasoning and to document the context for the future. 

For 2, this will help us clean up the rollkit docs. Right now, even for simple ignite apps, you have to run the local da, local sequencer, and in the future some local execution. This makes the docs and rollkit feel clunky. 

By adding the mock server commands to the intercept command, we can spin up mock da and mock sequencer for these tutorials to reduce the steps required by the users. 

Enables: https://github.com/rollkit/docs/issues/485
Closes #1910 

## Release Notes

- **New Features**
  - Added command-line option `--rollkit.sequencer_rollup_id` for specifying the sequencer middleware rollup ID, defaulting to "mock-rollup."
  - Introduced functionality to start mock services for improved testing and development.

- **Bug Fixes**
  - Enhanced error handling for starting mock servers, including checks for already running services.

- **Documentation**
  - Updated documentation to reflect new command-line options and improved configurations.

- **Refactor**
  - Streamlined control flow and error handling in the `InterceptCommand` and mock server initialization processes.
  - Improved code structure and readability for mock server startup logic.

### CODE REVIEW

File: cmd/rollkit/commands/run_node.go [hunk 1]
```diff
@@ -130,29 +128,36 @@ func NewRunNodeCmd() *cobra.Command {
...
 				}
-				// nolint:errcheck,gosec
-				defer func() { srv.Stop() }()
+			}()
+
+			// Determine which rollupID to use. If the flag has been set we want to use that value and ensure that the chainID in the genesis doc matches.
+			if cmd.Flags().Lookup(rollconf.FlagSequencerRollupID).Changed {
+				genDoc.ChainID = nodeConfig.SequencerRollupID
 			}
+			sequecnerRollupID := genDoc.ChainID
```
Comment (gupadhyaya) [inline cmd/rollkit/commands/run_node.go:148]: minor spellcheck

**Review by gupadhyaya — DISMISSED** [review]
lgtm

**Review by Manav-Aggarwal — APPROVED** [review]

File: cmd/rollkit/commands/intercept.go [hunk 2]
```diff
@@ -125,3 +165,14 @@ func RunRollupEntrypoint(rollkitConfig *rollconf.TomlConfig, args []string) erro
 
 	return nil
 }
+
+func parseFlag(args []string, flag string) string {
```
Comment (yarikbratashchuk) [inline cmd/rollkit/commands/intercept.go:169]: aren't we having primitives for flag parsing?

Comment (yarikbratashchuk) [reply cmd/rollkit/commands/intercept.go:169]: this doesn't take into account --flagname=value

**Review by yarikbratashchuk — COMMENTED** [review]
looks good
have you tested it?

**Review by tzdybal — APPROVED** [review]

File: cmd/rollkit/commands/intercept.go (hunk 2 above)
Comment (MSevey) [reply cmd/rollkit/commands/intercept.go:169]: @yarikbratashchuk REF: https://github.com/rollkit/rollkit/pull/1911#discussion_r1835021377

let me know if this addresses your issue. We can handle other questions offline as follow ups.

**Pull request merged**